---
title: "Supplementary Figures"
author: "Cong Wang"
date: "2023-06-24"
output: html_document
---

```{r message=FALSE, warning=FALSE}
library(ggplot2)
library(vegan)
library(colorRamps)
library(psych)
library(tidyr)
library(readxl)
library(ape)
library(ggplot2)
library(colorRamps)
library(ggrepel)

rm(list = ls())
load("data/CForBio.data.prep.2022.11.22.RData")

library(vegan)
cutoff <- 1000

sort(colSums(bac.amp1))
sort(colSums(bac.mgm1))
sort(colSums(ko.mgm1))
sort(colSums(cog.mgm1))
sort(colSums(rgi.mgm1))
sort(colSums(cazy.mgm1))

bac.amp <- rrarefy(t(bac.amp1),sample = 24903 )
bac.mgm <- data.frame(t(bac.mgm1))

ko.mgm <- data.frame(t(ko.mgm1))
cog.mgm <- data.frame(t(cog.mgm1))
rgi.mgm <- data.frame(t(rgi.mgm1))
cazy.mgm <- data.frame(t(cazy.mgm1))

env.amp1$rgi.mgm.H <- vegan::diversity(rgi.mgm)
env.amp1$cazy.mgm.H <- vegan::diversity(cazy.mgm)
env.amp1$ko.mgm.H <- vegan::diversity(ko.mgm)
env.amp1$cog.mgm.H <- vegan::diversity(cog.mgm)
env.amp1$bac.mgm.H <- vegan::diversity(bac.mgm)
env.amp1$bac.amp.H <- vegan::diversity(bac.amp)

env.amp1$rgi.mgm.S <- vegan::specnumber(rgi.mgm)
env.amp1$cazy.mgm.S <- vegan::specnumber(cazy.mgm)
env.amp1$ko.mgm.S <- vegan::specnumber(ko.mgm)
env.amp1$cog.mgm.S <- vegan::specnumber(cog.mgm)
env.amp1$bac.mgm.S <- vegan::specnumber(bac.mgm)
env.amp1$bac.amp.S <- vegan::specnumber(bac.amp)

env.amp1$cog.mgm.sum <- rowSums(cog.mgm)
env.amp1$rgi.mgm.sum <- rowSums(rgi.mgm)
env.amp1$cazy.mgm.sum <- rowSums(cazy.mgm)
env.amp1$ko.mgm.sum <- rowSums(ko.mgm)
env.amp1$bac.mgm.sum <- rowSums(bac.mgm)

#genomeTraits
genomeTrait<-read_excel("data/genomeTrait/genomesize.contigsprok/genomesize_novirus/all_genomesize.novirus.xlsx",sheet = "Sheet2")

genomeTrait$sample_name == env.amp1$sample_name

env.genomeTraits<-cbind(env.amp1,genomeTrait[,c(2:8)])

```

```{r message=FALSE, warning=FALSE}
#Supplementary Fig. 1
#ordination
library(ape)
library(ggplot2)
library(colorRamps)

#bac.amplicon
#calculate distance matrix
set.seed(315)
bac.amp.dist<-vegdist(bac.amp,method = 'bray')

#cmdscale function
bac.amp.pc<-cmdscale(bac.amp.dist,k=(nrow(bac.amp)-1),eig = TRUE)
eig<-bac.amp.pc$eig
bac.amp.pc1<-(eig[1]*100/sum(eig))
bac.amp.pc2<-(eig[2]*100/sum(eig))

#subset environmental data table
env.sub <- env.amp1[, c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich")]; 
colnames(env.sub) <-  c("Latitude","Longitude","Altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness")

#envfit analysis
ef<-envfit(bac.amp.pc,env.sub,permutations=999,na.rm=TRUE)
ef

#extract scores and add color var
pcoa.scores.bact.amp<-data.frame(bac.amp.pc$points)[1:35]
pcoa.scores.bact.amp$Site.latitude<-env.amp1$site.latitude
m1<-max(abs(pcoa.scores.bact.amp$X1)); m2<-max(abs(pcoa.scores.bact.amp$X2))

#extract envfit result
en_coord_cont.bact.amp<-as.data.frame(scores(ef,"vectors"))*ordiArrowMul(ef)
en_coord_cont.bact.amp
en_coord_cont.bact.amp$envfactor<-rownames(en_coord_cont.bact.amp)

p.bact = ggplot(data = pcoa.scores.bact.amp, aes(x = X1, y = X2)) + 
  geom_point(size=5,mapping = aes(color=Site.latitude))+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  geom_segment(aes(x = 0, y = 0, xend = Dim1*m1, yend = Dim2*m1),arrow = arrow(length = unit(0.02, "npc")),
               data = en_coord_cont.bact.amp, size =0.6, colour = "grey")+
  geom_text_repel(data = en_coord_cont.bact.amp, aes(x=Dim1*m1,y=Dim2*m1,label = envfactor),
                  colour="black",size=3)+
  ggtitle("Bacteria.16S.Amplicon")+
  labs(x = sprintf("PCo1 (%.1f%%)", bac.amp.pc1), y = sprintf("PCo2 (%.1f%%)", bac.amp.pc2))+
  theme_bw()+theme(axis.text.x = element_text(size = 12,color = "black"),axis.title.x = element_text(size = 12,color = "black"),
                   axis.text.y = element_text(size = 12,color = "black",),axis.title.y = element_text(size = 12,color = "black"),
                   legend.title = element_text(size=12))

p.bact

#ggsave("pdf2/Fig.S5a_2.PcoA.bact.composition.2023.08.24.pdf", width = 6, height = 4)

#bac.mgm
set.seed(315)
bac.mgm.dist<-vegdist(bac.mgm,method = 'bray')

#cmdscale function
bac.mgm.pc<-cmdscale(bac.mgm.dist,k=(nrow(bac.mgm)-1),eig = TRUE)
eig<-bac.mgm.pc$eig
bac.mgm.pc1<-(eig[1]*100/sum(eig))
bac.mgm.pc2<-(eig[2]*100/sum(eig))

#subset environmental data table
env.sub <- env.amp1[, c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich")]; 
colnames(env.sub) <-  c("Latitude","Longitude","Altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness")

#envfit analysis
ef<-envfit(bac.mgm.pc,env.sub,permutations=999,na.rm=TRUE)
ef

#extract scores and add color var
pcoa.scores.bact.mgm<-data.frame(bac.mgm.pc$points)[1:26]
pcoa.scores.bact.mgm$Site.latitude<-env.amp1$site.latitude
m1<-max(abs(pcoa.scores.bact.mgm$X1)); m2<-max(abs(pcoa.scores.bact.mgm$X2))


#extract envfit result
en_coord_cont.bact.mgm<-as.data.frame(scores(ef,"vectors"))*ordiArrowMul(ef)
en_coord_cont.bact.mgm
en_coord_cont.bact.mgm$envfactor<-rownames(en_coord_cont.bact.mgm)

p.bact.mgm = ggplot(data = pcoa.scores.bact.mgm, aes(x = X1, y = X2)) + 
  geom_point(size=5,mapping = aes(color=Site.latitude))+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  geom_segment(aes(x = 0, y = 0, xend = Dim1*m1, yend = Dim2*m2),arrow = arrow(length = unit(0.02, "npc")),
               data = en_coord_cont.bact.mgm, size =0.6, colour = "grey")+
  geom_text_repel(data = en_coord_cont.bact.mgm, aes(x=Dim1*m1,y=Dim2*m2,label = envfactor),
                  colour="black",size=3)+
  ggtitle("Bacteria.Metagenome.Shotgun")+
  labs(x = sprintf("PCo1 (%.1f%%)", bac.mgm.pc1), y = sprintf("PCo2 (%.1f%%)", bac.mgm.pc2))+
  theme_bw()+theme(axis.text.x = element_text(size = 12,color = "black"),axis.title.x = element_text(size = 12,color = "black"),
                   axis.text.y = element_text(size = 12,color = "black",),axis.title.y = element_text(size = 12,color = "black"),
                   legend.title = element_text(size=12))
p.bact.mgm

#ggsave("pdf2/Fig.S5b.PcoA.bact.composition.2023.08.24.pdf", width = 6, height = 4)

#ko.mgm
set.seed(315)
ko.mgm.dist<-vegdist(ko.mgm,method = 'bray')

#cmdscale function
ko.mgm.pc<-cmdscale(ko.mgm.dist,k=(nrow(ko.mgm)-1),eig = TRUE)
eig<-ko.mgm.pc$eig
ko.mgm.pc1<-(eig[1]*100/sum(eig))
ko.mgm.pc2<-(eig[2]*100/sum(eig))

#subset environmental data table
env.sub <- env.amp1[, c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich")]; 
colnames(env.sub) <-  c("Latitude","Longitude","Altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness")

#envfit analysis
ef<-envfit(ko.mgm.pc,env.sub,permutations=999,na.rm=TRUE)
ef

#extract scores and add color var
pcoa.scores.ko.mgm<-data.frame(ko.mgm.pc$points)[1:35]
pcoa.scores.ko.mgm$Site.latitude<-env.amp1$site.latitude
m1<-max(abs(pcoa.scores.ko.mgm$X1)); m2<-max(abs(pcoa.scores.ko.mgm$X2))


#extract envfit result
en_coord_cont.ko.mgm<-as.data.frame(scores(ef,"vectors"))*ordiArrowMul(ef)
en_coord_cont.ko.mgm
en_coord_cont.ko.mgm$envfactor<-rownames(en_coord_cont.ko.mgm)

p.ko.mgm = ggplot(data = pcoa.scores.ko.mgm, aes(x = X1, y = X2)) + 
  geom_point(size=5,mapping = aes(color=Site.latitude))+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  geom_segment(aes(x = 0, y = 0, xend = Dim1*m1, yend = Dim2*m2),arrow = arrow(length = unit(0.02, "npc")),
               data = en_coord_cont.ko.mgm, size =0.6, colour = "grey")+
  geom_text_repel(data = en_coord_cont.ko.mgm, aes(x=Dim1*m1,y=Dim2*m2,label = envfactor),
                  colour="black",size=3)+
  ggtitle("KO.Metagenome")+
  labs(x = sprintf("PCo1 (%.1f%%)", ko.mgm.pc1), y = sprintf("PCo2 (%.1f%%)", ko.mgm.pc2))+
  theme_bw()+theme(axis.text.x = element_text(size = 12,color = "black"),axis.title.x = element_text(size = 12,color = "black"),
                   axis.text.y = element_text(size = 12,color = "black",),axis.title.y = element_text(size = 12,color = "black"),
                   legend.title = element_text(size=12))
p.ko.mgm

#ggsave("pdf2/Fig.S5c.PcoA.KO.composition.2023.08.24.pdf", width = 6, height = 4)

#rgi.mgm
set.seed(315)
rgi.mgm.dist<-vegdist(rgi.mgm,method = 'bray')

#cmdscale function
rgi.mgm.pc<-cmdscale(rgi.mgm.dist,k=(nrow(rgi.mgm)-1),eig = TRUE)
eig<-rgi.mgm.pc$eig
rgi.mgm.pc1<-(eig[1]*100/sum(eig))
rgi.mgm.pc2<-(eig[2]*100/sum(eig))

#subset environmental data table
env.sub <- env.amp1[, c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich")]; 
colnames(env.sub) <-  c("Latitude","Longitude","Altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness")

#envfit analysis
ef<-envfit(rgi.mgm.pc,env.sub,permutations=999,na.rm=TRUE)
ef

#extract scores and add color var
pcoa.scores.rgi.mgm<-data.frame(rgi.mgm.pc$points)[1:31]
pcoa.scores.rgi.mgm$Site.latitude<-env.amp1$site.latitude
m1<-max(abs(pcoa.scores.rgi.mgm$X1)); m2<-max(abs(pcoa.scores.rgi.mgm$X2))

#extract envfit result
en_coord_cont.rgi.mgm<-as.data.frame(scores(ef,"vectors"))*ordiArrowMul(ef)
en_coord_cont.rgi.mgm
en_coord_cont.rgi.mgm$envfactor<-rownames(en_coord_cont.rgi.mgm)

p.rgi.mgm = ggplot(data = pcoa.scores.rgi.mgm, aes(x = X1, y = X2)) + 
  geom_point(size=5,mapping = aes(color=Site.latitude))+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  geom_segment(aes(x = 0, y = 0, xend = Dim1*m1, yend = Dim2*m2),arrow = arrow(length = unit(0.02, "npc")),
               data = en_coord_cont.rgi.mgm, size =0.6, colour = "grey")+
  geom_text_repel(data = en_coord_cont.rgi.mgm, aes(x=Dim1*m1,y=Dim2*m2,label = envfactor),
                  colour="black",size=3)+
  ggtitle("ARG.Metagenome")+
  labs(x = sprintf("PCo1 (%.1f%%)", rgi.mgm.pc1), y = sprintf("PCo2 (%.1f%%)", rgi.mgm.pc2))+
  theme_bw()+theme(axis.text.x = element_text(size = 12,color = "black"),axis.title.x = element_text(size = 12,color = "black"),
                   axis.text.y = element_text(size = 12,color = "black",),axis.title.y = element_text(size = 12,color = "black"),
                   legend.title = element_text(size=12))
p.rgi.mgm

#ggsave("pdf2/Fig.S5d.PcoA.ARG.composition.2023.08.24.pdf", width = 6, height = 4)

#cazy.mgm
set.seed(315)
cazy.mgm.dist<-vegdist(cazy.mgm,method = 'bray')

#cmdscale function
cazy.mgm.pc<-cmdscale(cazy.mgm.dist,k=(nrow(cazy.mgm)-1),eig = TRUE)
eig<-cazy.mgm.pc$eig
cazy.mgm.pc1<-(eig[1]*100/sum(eig))
cazy.mgm.pc2<-(eig[2]*100/sum(eig))

#subset environmental data table
env.sub <- env.amp1[, c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich")]; 
colnames(env.sub) <-  c("Latitude","Longitude","Altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness")

#envfit analysis
ef<-envfit(cazy.mgm.pc,env.sub,permutations=999,na.rm=TRUE)
ef

#extract scores and add color var
pcoa.scores.cazy.mgm<-data.frame(cazy.mgm.pc$points)[1:24]
pcoa.scores.cazy.mgm$Site.latitude<-env.amp1$site.latitude
m1<-max(abs(pcoa.scores.cazy.mgm$X1)); m2<-max(abs(pcoa.scores.cazy.mgm$X2))

#extract envfit result
en_coord_cont.cazy.mgm<-as.data.frame(scores(ef,"vectors"))*ordiArrowMul(ef)
en_coord_cont.cazy.mgm
en_coord_cont.cazy.mgm$envfactor<-rownames(en_coord_cont.cazy.mgm)

p.cazy.mgm = ggplot(data = pcoa.scores.cazy.mgm, aes(x = X1, y = X2)) + 
  geom_point(size=5,mapping = aes(color=Site.latitude))+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  geom_segment(aes(x = 0, y = 0, xend = Dim1*m1, yend = Dim2*m2),arrow = arrow(length = unit(0.02, "npc")),
               data = en_coord_cont.cazy.mgm, size =0.6, colour = "grey")+
  geom_text_repel(data = en_coord_cont.cazy.mgm, aes(x=Dim1*m1,y=Dim2*m2,label = envfactor),
                  colour="black",size=3)+
  ggtitle("CAZy.Metagenome")+
  labs(x = sprintf("PCo1 (%.1f%%)", cazy.mgm.pc1), y = sprintf("PCo2 (%.1f%%)", cazy.mgm.pc2))+
  theme_bw()+theme(axis.text.x = element_text(size = 12,color = "black"),axis.title.x = element_text(size = 12,color = "black"),
                   axis.text.y = element_text(size = 12,color = "black",),axis.title.y = element_text(size = 12,color = "black"),
                   legend.title = element_text(size=12))
p.cazy.mgm

#ggsave("pdf2/Fig.S5e.PcoA.CAZy.composition.2023.08.24.pdf", width = 6, height = 4)

```

```{r}
#Supplementary Fig. 2
library(readxl)
#genomesize of otu
genomesize_otu<-read_excel("data/otu_genome/calculate/genomesize_otu.xlsx")
genomesize_otu$sample_name == env.amp1$sample_name
genomesize_otu<-cbind(genomesize_otu,env.amp1[,c(29,82)])

library(ggplot2)
library(colorRamps)

summary(lm(AGS_otu~pH,genomesize_otu))
shapiro.test(residuals(lm(AGS_otu~pH,genomesize_otu)))

p15<-ggplot() + geom_jitter(data=genomesize_otu,  height = 0, aes(x= pH, y=AGS_otu, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=genomesize_otu,aes(x= pH, y=AGS_otu), method ="lm", col= "white")+
  labs(x = "pH",y = "Average Genome Size (Mbp)")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"),name="Site.latitude")+
  theme(legend.title = element_text(colour="black", size=12, face="bold"),
        legend.text = element_text(colour="black", size=12, face="bold"),
        axis.text=element_text(size=12,face="bold"),
        axis.title=element_text(size=12,face="bold"))+
  annotate("text",x=6.5,y=5.5, label="R2 = 0.151\nP = 0.01106",
           size=5,color="black")+ggtitle("Based on GTDB Annotation")
p15

summary(lm(Protein_count~AGS_otu,genomesize_otu))
shapiro.test(residuals(lm(Protein_count~AGS_otu,genomesize_otu)))

p17<-ggplot() + geom_jitter(data=genomesize_otu,  height = 0, aes(x= AGS_otu, y=Protein_count, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=genomesize_otu,aes(x= AGS_otu, y=Protein_count), method ="lm", col= "white")+
  labs(x = "Average Genome Size (Mbp)",y = "Protein Count")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"),name="Site.latitude")+
  theme(legend.title = element_text(colour="black", size=12, face="bold"),
        legend.text = element_text(colour="black", size=12, face="bold"),
        axis.text=element_text(size=12,face="bold"),
        axis.title=element_text(size=12,face="bold"))+
  annotate("text",x=5,y=5000, label="R2 = 0.975\nP < 2.2e-16",
           size=5,color="black")+ggtitle("Based on GTDB Annotation")
p17
```


```{r fig.height=3.5, fig.width=8}
#Supplementary Fig. 2
library(ggpubr)
ggarrange(p15,p17,ncol = 2,nrow = 1,
          labels = c("a","b"),common.legend = TRUE,legend = "right")

#ggsave("pdf/Fig.S3_genometraits_otu.pdf",plot = last_plot(),units = "in",height = 3.5,width = 8,dpi = 600)

```

```{r fig.height=7, fig.width=16}
#Supplementary Fig.3
data1<-data.frame(t(bac.amp))
ID<-data.frame(bac.amp.ID1)
ID$OTU.ID<-paste(ID$OTU,ID$phylum,sep = ".")
env.amp1$pH2<-paste("pH",round(env.amp1$pH,digits = 2),env.amp1$sample_name,sep = ".")

opf<-"genus"
Flev<-ID[,opf] 
fung.lev<-data.frame(aggregate(data1,by=list(Flev) , sum))
rownames(fung.lev)<-fung.lev[,1]; fung.lev<-fung.lev[,-1]
data2<-data.frame(t(fung.lev))
  
total<-apply(data2, 1, sum); 
fung.relabu<-data.frame(lapply(data2, function(x) {  x / total  }) )  # change the abundance data into relative abundance#
  
fung.L<-fung.relabu
env.L<-env.amp1
fung.L <- fung.L[,order(-colSums(fung.L))] ## 
bac.genus<-cbind(env.L,fung.L)

colnames(bac.genus)
bac.genus2<-bac.genus[,c(29,82,102:111)]

library(tidyverse)

bac.genus3<-gather(bac.genus2,key = "Genus",value = "Abundance",-"pH",-"site.latitude")
bac.genus3$Genus<-factor(bac.genus3$Genus,levels = c("Rhodoplanes","DA101","Bradyrhizobium",
                                                     "Candidatus_Solibacter","Candidatus_Xiphinematobacter",
                                                     "Candidatus_Koribacter","Burkholderia","Mycobacterium",
                                                     "Salinispora","Methylosinus"))

ggplot()+geom_point(data = bac.genus3,aes(x=pH,y=Abundance*100,color=site.latitude),size=3)+
  facet_wrap(.~Genus,nrow = 2,ncol = 5)+
  geom_smooth(data = bac.genus3,aes(x=pH,y=Abundance*100),method = "lm")+
  theme_bw()+ylab("Relative Abundance (%)")+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"),name="Site.latitude")+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"),
        strip.text = element_text(size = 12))

library(ggpubr)
#ggsave("otu_genome/Bacteria_top10genus2.pdf",width = 16,height = 7)

shapiro.test(residuals(lm(bac.genus2$Rhodoplanes~bac.genus2$pH)))
summary(lm(bac.genus2$Rhodoplanes~bac.genus2$pH))

shapiro.test(residuals(lm(log(bac.genus2$DA101)~bac.genus2$pH)))
summary(lm(log(bac.genus2$DA101)~bac.genus2$pH))

shapiro.test(residuals(lm(log(bac.genus2$Bradyrhizobium)~bac.genus2$pH)))
summary(lm(log(bac.genus2$Bradyrhizobium)~bac.genus2$pH))

shapiro.test(residuals(lm(log(bac.genus2$Candidatus_Solibacter)~bac.genus2$pH)))
summary(lm(log(bac.genus2$Candidatus_Solibacter)~bac.genus2$pH))

shapiro.test(residuals(lm(log(bac.genus2$Candidatus_Xiphinematobacter)~bac.genus2$pH)))
summary(lm(log(bac.genus2$Candidatus_Xiphinematobacter)~bac.genus2$pH))

shapiro.test(residuals(lm(log(bac.genus2$Candidatus_Koribacter)~bac.genus2$pH)))
summary(lm(log(bac.genus2$Candidatus_Koribacter)~bac.genus2$pH))

shapiro.test(residuals(lm(log(bac.genus2$Burkholderia)~bac.genus2$pH)))
summary(lm(log(bac.genus2$Burkholderia)~bac.genus2$pH))

shapiro.test(residuals(lm(log(bac.genus2$Mycobacterium)~bac.genus2$pH)))
summary(lm(log(bac.genus2$Mycobacterium)~bac.genus2$pH))

shapiro.test(residuals(lm(sqrt(bac.genus2$Salinispora)~bac.genus2$pH)))
summary(lm(sqrt(bac.genus2$Salinispora)~bac.genus2$pH))

shapiro.test(residuals(lm(log(bac.genus2$Methylosinus)~bac.genus2$pH)))
summary(lm(log(bac.genus2$Methylosinus)~bac.genus2$pH))

```


```{r}
#Suplementary Fig.4
data.genomesize.bahram<-read.csv("data/bahram/data.genomesize4.csv")

p1<-ggplot()+geom_jitter(data = data.genomesize.bahram,aes(x=pH,y=AGS),size=5)+
  geom_smooth(data = data.genomesize.bahram,aes(x=pH,y=AGS),method = "lm")+
  xlab("pH")+ylab("Average Genome Size (Mbp)")+theme_bw()+
  theme(axis.text = element_text(size = 12,color = "black",face = "bold"),
        axis.title = element_text(size = 12,color = "black",face = "bold"))+
  annotate("text",x=7,y=6, label="rho = -0.43 ***",
           size=5,color="black")+ggtitle("Based on GTDB Annotation")

p1

p6<-ggplot()+geom_jitter(data = data.genomesize.bahram,aes(x=pH,y=AGS2_mgm),size=5)+
  geom_smooth(data = data.genomesize.bahram,aes(x=pH,y=AGS2_mgm),method = "lm")+
  xlab("pH")+ylab("Average Genome Size (Mbp)")+theme_bw()+
  theme(axis.text = element_text(size = 12,color = "black",face = "bold"),
        axis.title = element_text(size = 12,color = "black",face = "bold"))+
  annotate("text",x=7,y=6, label="rho = -0.22 *",
           size=5,color="black")+ggtitle("Based on Metagenome")

p6
```


```{r fig.height=3.5, fig.width=8}
#Supplementary Fig. 4
library(ggpubr)
ggarrange(p6,p1,ncol = 2,nrow = 1,
          labels = c("a","b"),common.legend = TRUE,legend = "right")

#ggsave("pdf/Fig.S4_genometraits_bahram.pdf",plot = last_plot(),units = "in",height = 3.5,width = 8,dpi = 600)

```

```{r}
#Supplementary Fig. 5
summary(lm(env.genomeTraits$GC_percent~env.genomeTraits$AGS))
shapiro.test(residuals(lm(env.genomeTraits$GC_percent~env.genomeTraits$AGS)))

p12.1<-ggplot() + geom_jitter(data=env.genomeTraits,  height = 0, aes(x= AGS, y=GC_percent, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.genomeTraits,aes(x= AGS, y=GC_percent), method ="lm", col= "white")+
  labs(x = "Average Genome Size (Mbp)",y = "GC Content (%)")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"),name="Site.latitude")+
  theme(legend.title = element_text(colour="black", size=12, face="bold"),
        legend.text = element_text(colour="black", size=12, face="bold"),
        axis.text=element_text(size=12,face="bold"),
        axis.title=element_text(size=12,face="bold"))+
  annotate("text",x=3,y=59, label="R2 = 0.371\nP = 4.88e-05",
           size=5,color="black")+ggtitle("Based on Metagenome")
p12.1

summary(lm(GC_otu~AGS_otu,genomesize_otu))
shapiro.test(residuals(lm(GC_otu~AGS_otu,genomesize_otu)))

p19<-ggplot() + geom_jitter(data=genomesize_otu,  height = 0, aes(x= AGS_otu, y=GC_otu*100, color = site.latitude), alpha = 0.8, size =5) + 
  #geom_smooth(data=genomesize_otu,aes(x= AGS_otu, y=GC_otu*100), method ="lm", col= "white")+
  labs(x = "Average Genome Size (Mbp)",y = "GC Percent (%)")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"),name="Site.latitude")+
  theme(legend.title = element_text(colour="black", size=12, face="bold"),
        legend.text = element_text(colour="black", size=12, face="bold"),
        axis.text=element_text(size=12,face="bold"),
        axis.title=element_text(size=12,face="bold"))+
  annotate("text",x=5.4,y=60, label="R2 = 0.021\nP = 0.1962",size=5,color="black")+
  ggtitle("Based on GTDB Annotation")
p19
```


```{r fig.height=3.5, fig.width=8}
#Supplementary Fig. 5
library(ggpubr)
ggarrange(p12.1,p19,ncol = 2,nrow = 1,
          labels = c("a","b"),common.legend = TRUE,legend = "right")

#ggsave("pdf/Fig.S5-1_genometraits_bahram.pdf",plot = last_plot(),units = "in",height = 3.5,width = 9,dpi = 600)
```

```{r}
#Supplementary Fig. 6

summary(lm(gc_percent~AGS,data = data.genomesize.bahram))
shapiro.test(residuals(lm(gc_percent~AGS,data = data.genomesize.bahram)))

p10<-ggplot()+geom_jitter(data = data.genomesize.bahram,aes(x=AGS2_mgm,y=GC_percent_mgm),size=5)+
  geom_smooth(data = data.genomesize.bahram,aes(x=AGS2_mgm,y=GC_percent_mgm),method = "lm")+
  ylab("GC Percent (%)")+xlab("Average Genome Size (Mbp)")+theme_bw()+
  theme(axis.text = element_text(size = 12,color = "black",face = "bold"),
        axis.title = element_text(size = 12,color = "black",face = "bold"))+
  annotate("text",x=5,y=67, label="rho = -0.57 ***",
           size=5,color="black")+ggtitle("Based on Metagenome")

p10

summary(lm(gc_percent~AGS,data = data.genomesize.bahram))
shapiro.test(residuals(lm(gc_percent~AGS,data = data.genomesize.bahram)))

p5<-ggplot()+geom_jitter(data = data.genomesize.bahram,aes(x=AGS,y=gc_percent*100),size=5)+
  geom_smooth(data = data.genomesize.bahram,aes(x=AGS,y=gc_percent*100),method = "lm")+
  ylab("GC Percent (%)")+xlab("Average Genome Size (Mbp)")+theme_bw()+
  theme(axis.text = element_text(size = 12,color = "black",face = "bold"),
        axis.title = element_text(size = 12,color = "black",face = "bold"))+
  annotate("text",x=5,y=65, label="rho = 0.25 **",
           size=5,color="black")+ggtitle("Based on GTDB Annotation")

p5
```


```{r}
#Supplementary Fig. 6
library(ggpubr)
ggarrange(p10,p5,ncol = 2,nrow = 1,
          labels = c("a","b"),common.legend = TRUE,legend = "right")

#ggsave("pdf/Fig.S5-2_genometraits_bahram.pdf",plot = last_plot(),units = "in",height = 3.5,width = 8,dpi = 600)

```

```{r message=FALSE, warning=FALSE}
#Supplementary FIg.7
p1<-ggplot() + geom_jitter(data=env.amp1,  height = 0, aes(x= pH, y=bac.amp.H, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.amp1,aes(x= pH, y=bac.amp.H), method ="lm", col= "white")+
  labs(x = "Soil pH",y = "H'.16S")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6.5,y=6.3, label="R2 = 0.481\nP = 1.653e-06",
           size=5,color="black")
p1
#ggsave("pdf/Fig.1a.corr.pH.H16S.2023.04.18.pdf", width = 4, height = 3)
summary(lm(env.amp1$bac.amp.H~env.amp1$pH))
shapiro.test(residuals(lm(env.amp1$bac.amp.H~env.amp1$pH)))

p3<-ggplot() + geom_jitter(data=env.amp1,  height = 0, aes(x= pH, y=ko.mgm.H, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.amp1,aes(x= pH, y=ko.mgm.H), col= "white")+
  labs(x = "Soil pH",y = "H'.KO")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6.5,y=7.9, label="R2 = 0.199\nP =  0.009",
           size=5,color="black")
p3
#ggsave("pdf/Fig.1c.corr.pH.HKO.2022.12.29.pdf", width = 4, height = 3)
m2<-lm(env.amp1$ko.mgm.H ~ env.amp1$pH +  I(env.amp1$pH^2)  )
AIC(m2)
summary(m2)
shapiro.test(residuals(m2))

p5<-ggplot() + geom_jitter(data=env.amp1,  height = 0, aes(x= pH, y = rgi.mgm.H, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.amp1,aes(x= pH, y=rgi.mgm.H), method = "lm", color = "white")+
  labs(x = "Soil pH",y = "H'.ARG")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6.5,y=4.9, label="R2 = 0.190\nP = 0.004",
           size=5,color="black")
p5

summary(lm(env.amp1$rgi.mgm.H~env.amp1$pH))
shapiro.test(residuals(lm(env.amp1$rgi.mgm.H~env.amp1$pH)))
#ggsave("pdf2/Fig.2b.corr.pH.HARG.2022.12.29.pdf", width = 4, height = 3)

p7<-ggplot() + geom_jitter(data=env.amp1,  height = 0, aes(x= pH, y = cazy.mgm.H, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.amp1,aes(x= pH, y=cazy.mgm.H), method = "lm", color = "white")+
  labs(x = "Soil pH",y = "H'.Cazy")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6.5,y=3.55, label="R2 = 0.833\nP = 5.52e-15",
           size=5,color="black")
p7

summary(lm(env.amp1$cazy.mgm.H~env.amp1$pH))
shapiro.test(residuals(lm(env.amp1$cazy.mgm.H~env.amp1$pH)))
#ggsave("pdf2/Fig.2c.corr.pH.HCazy.2022.12.29.pdf", width = 4, height = 3)

p1.1<-ggplot() + geom_jitter(data=env.amp1,  height = 0, aes(x= round(ko.mgm.H,digits = 2), y=bac.amp.H, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.amp1,aes(x= ko.mgm.H, y=bac.amp.H), method ="lm", col= "white")+
  labs(x = "H'.KO",y = "H'.16S")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=7.87,y=7.2, label="R2 = 0.294\nP = 0.00037",
           size=5,color="black")
p1.1
#ggsave("pdf/Fig.1b.corr.pH.H16S.2023.04.18.pdf", width = 4, height = 3)
summary(lm(env.amp1$bac.amp.H~env.amp1$ko.mgm.H))
shapiro.test(residuals(lm(env.amp1$bac.amp.H~env.amp1$ko.mgm.H)))

p3.1<-ggplot() + geom_jitter(data=env.genomeTraits,  height = 0, aes(x= AGS, y=round(ko.mgm.H,digits = 2), color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.genomeTraits,aes(x= AGS, y=ko.mgm.H), method = "lm",col= "white")+
  labs(x = "Average Genome Size (Mbp)",y = "H'.KO")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6,y=7.89, label="R2 = 0.0006\nP = 0.886",
           size=5,color="black")

p3.1

#ggsave("pdf2/Fig.2a.corr.pH.HKO.2022.12.29.pdf", width = 4, height = 3)
summary(lm(env.genomeTraits$ko.mgm.H~env.genomeTraits$AGS))
shapiro.test(residuals(lm(env.genomeTraits$ko.mgm.H~env.genomeTraits$AGS)))

p5.1<-ggplot() + geom_jitter(data=env.genomeTraits,  height = 0, aes(x= AGS, y=round(rgi.mgm.H,digits = 2), color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.genomeTraits,aes(x= AGS, y=rgi.mgm.H), method = "lm",col= "white")+
  labs(x = "Average Genome Size (Mbp)",y = "H'.ARG")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6,y=4.9, label="R2 = 0.239\nP = 0.001",
           size=5,color="black")
p5.1

#ggsave("pdf2/Fig.2a.corr.pH.HKO.2022.12.29.pdf", width = 4, height = 3)
summary(lm(env.genomeTraits$rgi.mgm.H~env.genomeTraits$AGS))
shapiro.test(residuals(lm(env.genomeTraits$rgi.mgm.H~env.genomeTraits$AGS)))

p7.1<-ggplot() + geom_jitter(data=env.genomeTraits,  height = 0, aes(x= AGS, y=cazy.mgm.H, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.genomeTraits,aes(x= AGS, y=cazy.mgm.H), method = "lm",col= "white")+
  labs(x = "Average Genome Size (Mbp)",y = "H'.Cazy")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6,y=3.6, label="R2 = 0.322\nP = 0.0001",
           size=5,color="black")
p7.1
#ggsave("pdf2/Fig.2a.corr.pH.HKO.2022.12.29.pdf", width = 4, height = 3)
summary(lm(env.genomeTraits$cazy.mgm.H~env.genomeTraits$AGS))
shapiro.test(residuals(lm(env.genomeTraits$cazy.mgm.H~env.genomeTraits$AGS)))
```

```{r fig.height=7, fig.width=16, message=FALSE, warning=FALSE}
#Supplementary FIg.7
library(ggpubr)
ggarrange(p1,p3,p5,p7,
          p1.1,p3.1,p5.1,p7.1,
          ncol = 4,nrow = 2,widths=c(1,1),heights=c(1,1),common.legend = TRUE,legend = "right",
          labels = c("a","b","c","d","e","f","g","h"))

#ggsave("pdf/Supplementary.Fig2_richness&diversity.pdf",plot = last_plot(),units = "in",height = 7,width = 16,dpi = 600)

```

```{r fig.height=10, fig.width=15, message=FALSE, warning=FALSE}
#Supplementary Fig. 8
library(linkET)
library(dplyr)

colnames(bac.amp1) == colnames(ko.mgm1)

bac.amp2<-data.frame(t(bac.amp1))
bac.amp2<-data.frame(apply(bac.amp2,1,function(x){x/sum(x)}))

ko.mgm2<-data.frame(t(ko.mgm1))
ko.mgm2<-data.frame(apply(ko.mgm2,1,function(x){x/sum(x)}))

bac.ko<-cbind(data.frame(t(bac.amp1)),data.frame(t(ko.mgm1)))

env.mantel<-env.genomeTraits[,c("AGS","ACN","GC_percent","Growthrate",
                                "bac.amp.S","bac.amp.H","ko.mgm.S","ko.mgm.H",
                                "latitude","longitude","altitude","MAP","MAT",
                                "TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N",
                                "C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich")]

colnames(env.mantel)<-c("AGS","ACN","GC content","Doubling time",
                        "S.16S","H'.16S","S.KO","H'.KO",
                        "Latitude","Longitude","Altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe", "AK","C_N","C_P", "N_P", "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness")
#plot
mantel <- mantel_test(bac.ko, env.mantel,
                      spec_select = list(BacteriaCom = 1:22579,
                                         FunctionCom = 22580:33644),
                      spec_dist =  dist_func(.FUN = "vegdist", method = "bray"),
                      env_dist = dist_func(.FUN = "vegdist", method = "euclidean")) %>% 
  mutate(rd = cut(r, breaks = c(-Inf, 0.2, 0.4, Inf),
                  labels = c("< 0.2", "0.2 - 0.4", ">= 0.4")),
         pd = cut(p, breaks = c(-Inf, 0.01, 0.05, Inf),
                  labels = c("< 0.01", "0.01 - 0.05", ">= 0.05")))

qcorrplot(correlate(env.mantel,method = "spearman"),type = "lower", diag = FALSE,
          grid_size = 0.25) +
  geom_square() +
  geom_mark(sep = '\n',size=3,sig_level = c(0.05,0.01,0.001),sig_thres = 0.05)+
  geom_couple(aes(colour = pd, size = rd), data = mantel, curvature = 0.1) +
  scale_fill_gradientn(colours = RColorBrewer::brewer.pal(11, "RdYlBu")) +
  #scale_fill_gradient(guide = "legend", high='#2c7bb6', low='#d7191c',name="rho")+
  scale_size_manual(values = c(0.5, 1, 2)) +
  scale_colour_manual(values = c("#f6e8c3", "#c7eae5", "#A2A2A288")) +
  #scale_colour_manual(values = color_pal(3, 0)) +
  guides(size = guide_legend(title = "Mantel's r",
                             override.aes = list(colour = "grey35"), 
                             order = 2),
         colour = guide_legend(title = "Mantel's p", 
                               override.aes = list(size = 3), 
                               order = 1),
         fill = guide_colorbar(title = "Spearman's rho", order = 3))

```

```{r}
#Supplementary Fig. 9
p1<-ggplot() + geom_jitter(data=env.amp1,  height = 0, aes(x= pH, y=bac.mgm.H, color = site.latitude), alpha = 0.8, size =5) + 
  #geom_smooth(data=env.amp1,aes(x= pH, y=bac.mgm.H), method ="lm", col= "white")+
  labs(x = "Soil pH",y = "H'.Bacteria.mgm")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6.5,y=6.3, label="R2 = 0.009\nP = 0.253",
           size=5,color="black")
p1

#ggsave("pdf2/Fig.S2A.corr.pH.H16S.2023.04.18.pdf", width = 6, height = 4)
summary(lm(env.amp1$bac.mgm.H~env.amp1$pH))
shapiro.test(residuals(lm(env.amp1$bac.mgm.H~env.amp1$pH)))

library(lme4)
library(lmerTest)
library(sjPlot)
lm<-lmer(bac.amp.H~1+pH+(1|latitude),data = env.amp1,
         REML = FALSE,control=lmerControl(optimizer="bobyqa"))

shapiro.test(residuals(lm))
summary(lm)
tab_model(lm)


p2<-ggplot() + geom_jitter(data=env.amp1,  height = 0, aes(x= pH, y=bac.mgm.S, color = site.latitude), alpha = 0.8, size =5) + 
  #geom_smooth(data=env.amp1,aes(x= pH, y=bac.mgm.S), method ="loess", col= "white")+
  labs(x = "Soil pH",y = "S.Bacteria.mgm")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=6.5,y=33500, label="R2 = 0.0184\nP = 0.429",
           size=5,color="black")
p2
#ggsave("pdf2/Fig.S2B.corr.pH.H16S.2023.04.18.pdf", width = 6, height = 4)
summary(lm(env.amp1$bac.mgm.S~env.amp1$pH))
shapiro.test(residuals(lm(env.amp1$bac.mgm.S~env.amp1$pH)))

```

```{r message=FALSE, warning=FALSE}
#Supplementary Fig. 10
env.amp1$sample_name==env.genomeTraits$sample_name

env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                     "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                     "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(env.amp1[,c("bac.mgm.H","bac.mgm.S","bac.amp.H","bac.amp.S", "ko.mgm.H","ko.mgm.S","rgi.mgm.H","rgi.mgm.S", "cazy.mgm.H","cazy.mgm.S")])

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r$X<-row.names(r)
r.long <- gather(r, Y, r, 1 : ncol(r)-1, factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(r)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(cor.out$r, 2)


cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))

cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

cor.out$Y <- factor(cor.out$Y, levels = c("bac.mgm.H","bac.mgm.S","bac.amp.H","bac.amp.S", "ko.mgm.H","ko.mgm.S","rgi.mgm.H","rgi.mgm.S", "cazy.mgm.H","cazy.mgm.S"))

cor.out$Y <- factor(cor.out$Y, labels = c("H'.bacteria.mgm","S.bacteria.mgm","H'.16S","S.16S","H'.KO","S.KO","H'.ARG","S.ARG","H'.Cazy","S.Cazy"))

ggplot(cor.out, aes(Y,X)) + 
  geom_tile(aes(fill = r), size=1)+
  geom_text(aes(label = r), size = 1.5)+
  #scale_fill_gradient(guide = "legend", high='green', low='blue')+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  theme(axis.title= element_blank(),
        axis.text.x=element_text(colour="black", size=8, face="bold",angle = 20),
        axis.text.y=element_text(colour="black", size=8, face="bold"))

cor.out.sig <- cor.out
cor.out.sig$r[cor.out.sig$p > 0.01] <- 0

p4<-ggplot(cor.out.sig, aes(Y,X)) + 
  geom_tile(aes(fill = r), size=1)+
  geom_text(data=  cor.out.sig[cor.out.sig$r!=0,], aes(label = r), size = 2.5, font="bold")+theme_bw()+
  #scale_fill_gradient(guide = "legend", high='green', low='blue',name="rho")+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
   geom_hline(yintercept = c(8.5,9.5), color = "orange")+
   theme(axis.title= element_blank(),
        axis.text.x=element_text(colour="black", size=12, face="bold",angle = 90, hjust=1,vjust = 0.5),
        axis.text.y=element_text(colour="black", size=12, face="bold"))
p4

#ggsave("pdf/Supplementary_Fig.9.pdf", width = 8, height = 6)

```


```{r}
#Supplementary Fig. 11
p1<-ggplot() + geom_jitter(data=env.genomeTraits,  height = 0, aes( x=bac.amp.S,y= AGS, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.genomeTraits,aes(y= AGS, x=bac.amp.S), method ="lm", col= "white")+
  labs(y = "Average Genome Size (Mbp)",x = "S.16S")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=4000,y=6.5, label="R2 = 0.187\nP = 0.0049",
           size=5,color="black")
p1

summary(lm(env.genomeTraits$AGS~env.genomeTraits$bac.amp.S))
shapiro.test(residuals(lm(env.genomeTraits$AGS~env.genomeTraits$bac.amp.S)))

p2<-ggplot() + geom_jitter(data=env.genomeTraits,  height = 0, aes( x=bac.amp.H,y= AGS, color = site.latitude), alpha = 0.8, size =5) + 
  geom_smooth(data=env.genomeTraits,aes(y= AGS, x=bac.amp.H), method ="lm", col= "white")+
  labs(y = "Average Genome Size (Mbp)",x = "H'.16S")+
  theme_bw()+
  scale_colour_manual(values= c(blue2red(15)[1:5], blue2red(17)[11:15], "green", "black"))+
  theme(legend.title = element_text(colour="black", size=15, face="bold"),
        legend.text = element_text(colour="black", size=15, face="bold"),
        axis.text=element_text(size=15,face="bold"),
        axis.title=element_text(size=15,face="bold"))+
  annotate("text",x=7,y=6.5, label="R2 = 0.222\nP = 0.0021",
           size=5,color="black")
p2

summary(lm(env.genomeTraits$AGS~env.genomeTraits$bac.amp.H))
shapiro.test(residuals(lm(env.genomeTraits$AGS~env.genomeTraits$bac.amp.H)))
```


```{r fig.height=3.5, fig.width=8}
#Supplementary Fig. 11
library(ggpubr)
ggarrange(p1,p2,
          ncol = 2,nrow = 1,widths=c(1,1),heights=c(1,1),common.legend = TRUE,legend = "right",
          labels = c("a","b"))

#ggsave("pdf/Supplementary.Fig.3_16S&AGS.pdf",plot = last_plot(),units = "in",height = 3.5,width = 8,dpi = 600)

```

```{r fig.height=4, fig.width=10, message=FALSE, warning=FALSE}
#Supplementary Fig. 12a
env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(cog.mgm1))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))
cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

library(splitstackshape)

cor.out.sig <- cor.out
cor.out.sig$r[cor.out.sig$p > 0.01] <- 0
cor.out.sig$Y_2<-"COGs"

ggplot(cor.out.sig, aes(Y,X)) +
  facet_grid(.~ Y_2,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  geom_text(data=  cor.out.sig[cor.out.sig$r!=0,], aes(label = r), size = 2, font="bold")+scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
   geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(axis.title= element_blank(),
        strip.text = element_text(colour="black",size = 12),
        axis.text.x=element_text(colour="black", size=12, face="bold"),
        axis.text.y=element_text(colour="black", size=12, face="bold"))

#ggsave("pdf/Supplementary.Fig10a.env.cog.genes.2023.01.05.pdf", width = 10, height = 4)

```


```{r fig.height=5, fig.width=20, message=FALSE, warning=FALSE}
#Supplementary Fig. 12b
env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.mgm1))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))

cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

library(splitstackshape)

cor.out.sig <- cor.out
cor.out.sig$r[cor.out.sig$p > 0.01] <- 0
cor.out.sig$Y_2<-"All annotated KOs"
  
ggplot(cor.out.sig, aes(Y,X)) +
  facet_grid(.~ Y_2,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
   geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 12,face="bold", color = c("black")),
        #strip.background = element_rect(fill = "grey"),
        panel.spacing = unit(0.1, "lines"),
        axis.title= element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y=element_text(colour="black", size=15, face="bold"))

#ggsave("pdf/Supplementary.Fig10b.corrHeatmap.env.ko.genes.2023.01.05.pdf", width = 20, height = 5)

```

```{r fig.height=7, fig.width=30, message=FALSE, warning=FALSE}
#Supplementary Fig. 13
ko.ID.sub<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "r.k_4")
ko.ID.sub <- merge(ko.ID.sub, ko.mgm.ID0, by.x = "KO", by.y = "Category", all = F)
ko.mgm1$KO <- row.names(ko.mgm1)
ko.mgm.sub0 <- merge(ko.ID.sub,ko.mgm1, by = "KO", all = T )
ko.mgm.sub0 <- ko.mgm.sub0[!is.na(ko.mgm.sub0$Level1),]
ko.mgm.sub0 <- ko.mgm.sub0[ order(ko.mgm.sub0$No),]
row.names(ko.mgm.sub0) <- paste0(ko.mgm.sub0$KO,"_", ko.mgm.sub0$No, "_", ko.mgm.sub0$Level1, "_", ko.mgm.sub0$Level2 , "_", ko.mgm.sub0$Level3)

ko.sel <-ko.mgm.sub0[, c(-1:-6)]
ko.sel.ID <- ko.mgm.sub0[, c(1:6)]
colnames(ko.sel) == env.amp1$sample_name


env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.sel))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))

cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

library(splitstackshape)

cor.out<-cSplit(cor.out, "Y", "_")

cor.out$Y_4 <- factor(cor.out$Y_4, levels = c("Bacterial.secretion.system","Cell.motility","Xenobiotics.biodegradation.and.metabolism","Signal.transduction","Metabolism.of.terpenoids.and.polyketides","Glycan.biosynthesis.and.metabolism","Porphyrin.metabolism",
                                              "Energy.metabolism","Membrane.transport","Citrate.cycle","Glyoxylate.and.dicarboxylate.metabolism","Metabolism.of.other.amino.acids"))

cor.out$Y_4 <- factor(cor.out$Y_4,labels = c("BSS","CM","XenoBDM","Two component system","MTP","GBM","PM",
                                             "Energy metabolism","Membrane transport","CC","GDM","MAC"))


cor.out.sub<-cor.out[cor.out$Y_4=="GBM",]
levels(droplevels(factor(cor.out.sub$Y_5)))

cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, levels = c( "Arabinogalactan.biosynthesis.Mycobacterium", "Exopolysaccharide.biosynthesis", "Glycosaminoglycan.biosynthesis.chondroitin.sulfate.dermatan.sulfate", 
                                                       "Glycosaminoglycan.biosynthesis.heparan.sulfate.heparin", "Glycosaminoglycan.degradation", "Glycosphingolipid.biosynthesis.ganglio.series", 
                                                       "Glycosphingolipid.biosynthesis.globo.and.isoglobo.series", "Glycosphingolipid.biosynthesis.lacto.and.neolacto.series", "Glycosylphosphatidylinositol.anchor.biosynthesis", 
                                                       "Lipoarabinomannan.biosynthesis", "Lipopolysaccharide.biosynthesis", "O.Antigen.nucleotide.sugar.biosynthesis",  
                                                       "O.Antigen.repeat.unit.biosynthesis", "Mannose.type.O.glycan.biosynthesis", 
                                                       "N.Glycan.biosynthesis","Peptidoglycan.biosynthesis","Other.glycan.degradation", "Other.types.of.O.glycan.biosynthesis", "Various.types.of.N.glycan.biosynthesis", 
                                                        "Teichoic.acid.biosynthesis"))

cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, labels = c( "ABM", "Exopolysaccharide.biosynthesis", "C", 
                                                       "H", "GD", "S", 
                                                       "GI", "L", "A", 
                                                       "Lipoarabinomannan", "Lipopolysaccharide.biosynthesis", "O.Antigen.nucleotide.sugar.biosynthesis", "ARUB","M.G", 
                                                       "N.Glycan", "Peptidoglycan.biosynthesis", 
                                                       "Other.glycan.D", "O", "VN.glycanB", 
                                                       "Teichoic.acid.biosyn"))

p1<-ggplot(cor.out.sub, aes(Y_1,X)) +
  facet_grid(.~ Y_5,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 10,face="bold", color = c("white")),
        strip.background = element_rect(fill = "red"),
        panel.spacing = unit(0.1, "lines"), axis.title= element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y=element_text(colour="black", size=18, face="bold")) +
  labs(caption="ABM: Arabinogalactan.biosynthesis.Mycobacterium;C: Glycosaminoglycan.biosynthesis.chondroitin.sulfate.dermatan.sulfate; H: Glycosaminoglycan.biosynthesis.heparan.sulfate.heparin; GD: Glycosaminoglycan.degradation; S: Glycosphingolipid.biosynthesis.ganglio.series; GI: Glycosphingolipid.biosynthesis.globo.and.isoglobo.series; L: Glycosphingolipid.biosynthesis.lacto.and.neolacto.series; 
       A: Glycosylphosphatidylinositol.anchor.biosynthesis; Lipoarabinomannan: Lipoarabinomannan.biosynthesis;ARUB: O.Antigen.repeat.unit.biosynthesis; M.G: Mannose.type.O.glycan.biosynthesis; N.Glycan: N.Glycan.biosynthesis; Other.glycan.D: Other.glycan.degradation; O: Other.types.of.O.glycan.biosynthesis; VN.glycanB: Various.types.of.N.glycan.biosynthesis,Teichoic.acid.biosyn: Teichoic.acid.biosynthesis" )
p1
#ggsave("pdf2/Fig.S6a.corrHeatmap.env.Glycan.2023.01.05.pdf", width = 30, height = 7)

env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(cazy.mgm1))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)

r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))

cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

library(splitstackshape)

cor.out.sig <- cor.out
cor.out.sig$r[cor.out.sig$p > 0.01] <- 0
cor.out.sig$Y_2<-"Carbohydrate-active enzymes (CAZy)"

p2<-ggplot(cor.out.sig, aes(Y,X)) +
  facet_grid(.~ Y_2,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  geom_text(data=  cor.out.sig[cor.out.sig$r!=0,], aes(label = r), size = 1.5, font="bold")+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 12,face="bold", color = c("white")),
        strip.background = element_rect(fill = "red"),
        axis.title= element_blank(),
        axis.text.x=element_text(colour="black", size=12, face="bold",angle = 90, hjust=1,vjust = 0.5),
        axis.text.y=element_text(colour="black", size=18, face="bold"))
p2
#ggsave("pdf2/Fig.S6b.corrHeatmap.env.cazy.genes.2023.01.05.pdf", width = 20, height = 6)
```


```{r fig.height=13, fig.width=30, message=FALSE, warning=FALSE}
#Supplementary Fig. 13
library(ggpubr)
ggarrange(p1,p2,
          ncol = 1,nrow = 2,widths=c(1,1),heights=c(1,1),common.legend = TRUE,legend = "right",
          labels = c("a","b"))
#ggsave("pdf/Supplementary.Fig.S12.pdf",plot = last_plot(),units = "in",height = 13,width = 30,dpi = 600)

```

```{r fig.height=6, fig.width=20, message=FALSE, warning=FALSE}
#Supplementary Fig. 14
ko.ID.sub<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "r.k_4")
ko.ID.sub <- merge(ko.ID.sub, ko.mgm.ID0, by.x = "KO", by.y = "Category", all = F)
ko.mgm1$KO <- row.names(ko.mgm1)
ko.mgm.sub0 <- merge(ko.ID.sub,ko.mgm1, by = "KO", all = T )
ko.mgm.sub0 <- ko.mgm.sub0[!is.na(ko.mgm.sub0$Level1),]
ko.mgm.sub0 <- ko.mgm.sub0[ order(ko.mgm.sub0$No),]
row.names(ko.mgm.sub0) <- paste0(ko.mgm.sub0$KO,"_", ko.mgm.sub0$No, "_", ko.mgm.sub0$Level1, "_", ko.mgm.sub0$Level2 , "_", ko.mgm.sub0$Level3)


ko.sel <-ko.mgm.sub0[, c(-1:-6)]
ko.sel.ID <- ko.mgm.sub0[, c(1:6)]
colnames(ko.sel) == env.amp1$sample_name


env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.sel))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))

cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

library(splitstackshape)

cor.out<-cSplit(cor.out, "Y", "_")

cor.out$Y_4 <- factor(cor.out$Y_4, levels = c("Bacterial.secretion.system","Cell.motility","Xenobiotics.biodegradation.and.metabolism","Signal.transduction","Metabolism.of.terpenoids.and.polyketides","Glycan.biosynthesis.and.metabolism","Porphyrin.metabolism",
                                              "Energy.metabolism","Membrane.transport","Citrate.cycle","Glyoxylate.and.dicarboxylate.metabolism","Metabolism.of.other.amino.acids"))

cor.out$Y_4 <- factor(cor.out$Y_4,labels = c("BSS","CM","XenoBDM","Two component system","MTP","GBM","PM",
                                             "Energy metabolism","Membrane transport","CC","GDM","MAC"))


cor.out.sub<-cor.out[cor.out$Y_4=="CM",]

levels(droplevels(factor(cor.out.sub$Y_5)))

cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, labels = c("Bacterial chemotaxis", "Flagellar assembly","RAC"))

p1<- ggplot(cor.out.sub, aes(Y_1,X)) +
  facet_grid(.~ Y_5,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 10,face="bold", color = c("white")),
        strip.background = element_rect(fill = "red"),
        panel.spacing = unit(0.1, "lines"), axis.title= element_blank(),
        axis.text.x=element_text(colour="black", size=10, face="bold",angle = 90, hjust=1,vjust = 0.5),
        axis.text.y=element_text(colour="black", size=15, face="bold")) +
  labs(caption="RAC: Regulation of actin cytoskeleton")

p1

#ggsave("pdf2/Fig.S7a.Heatmap.KEGG.cellular.motility.2023.01.05.pdf", width = 20, height = 6)

ko.ID.sub<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "r.k_4")
ko.ID.sub <- merge(ko.ID.sub, ko.mgm.ID0, by.x = "KO", by.y = "Category", all = F)
ko.mgm1$KO <- row.names(ko.mgm1)
ko.mgm.sub0 <- merge(ko.ID.sub,ko.mgm1, by = "KO", all = T )
ko.mgm.sub0 <- ko.mgm.sub0[!is.na(ko.mgm.sub0$Level1),]
ko.mgm.sub0 <- ko.mgm.sub0[ order(ko.mgm.sub0$No),]
row.names(ko.mgm.sub0) <- paste0(ko.mgm.sub0$KO,"_", ko.mgm.sub0$No, "_", ko.mgm.sub0$Level1, "_", ko.mgm.sub0$Level2 , "_", ko.mgm.sub0$Level3)

ko.sel <-ko.mgm.sub0[, c(-1:-6)]
ko.sel.ID <- ko.mgm.sub0[, c(1:6)]
colnames(ko.sel) == env.amp1$sample_name

env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.sel))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))
cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

library(splitstackshape)

cor.out<-cSplit(cor.out, "Y", "_")

cor.out$Y_4 <- factor(cor.out$Y_4, levels = c("Bacterial.secretion.system","Cell.motility","Xenobiotics.biodegradation.and.metabolism","Signal.transduction","Metabolism.of.terpenoids.and.polyketides","Glycan.biosynthesis.and.metabolism","Porphyrin.metabolism",
                                              "Energy.metabolism","Membrane.transport","Citrate.cycle","Glyoxylate.and.dicarboxylate.metabolism","Metabolism.of.other.amino.acids"))

cor.out$Y_4 <- factor(cor.out$Y_4,labels = c("BSS","CM","XenoBDM","Two component system","MTP","GBM","PM",
                                             "Energy metabolism","Membrane transport","CC","GDM","MAC"))


cor.out.sub<-cor.out[cor.out$Y_4=="Two component system",]
levels(droplevels(factor(cor.out.sub$Y_5)))
#cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, labels = c("Bacterial chemotaxis", "Flagellar assembly","RAC"))

p2<-ggplot(cor.out.sub, aes(Y_1,X)) +
 facet_grid(.~ Y_5,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 12,face="bold",color = c("white")),
        strip.background = element_rect(fill = "red"),
        panel.spacing = unit(0.1, "lines"), axis.title= element_blank(),
        axis.text.x=element_text(colour="black", size=4, face="bold",angle = 90, hjust=1,vjust = 0.5),
        axis.text.y=element_text(colour="black", size=15, face="bold")) 
p2
#ggsave("pdf2/Fig.S7b.corrHeatmap.env.Signal.transduction.2023.01.05.pdf", width = 20, height = 6)

ko.ID.sub<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "r.k")
ko.ID.sub <- merge(ko.ID.sub, ko.mgm.ID0, by.x = "KO", by.y = "Category", all = F)
ko.mgm1$KO <- row.names(ko.mgm1)
ko.mgm.sub0 <- merge(ko.ID.sub,ko.mgm1, by = "KO", all = T )
ko.mgm.sub0 <- ko.mgm.sub0[!is.na(ko.mgm.sub0$Level1),]
ko.mgm.sub0 <- ko.mgm.sub0[ order(ko.mgm.sub0$No),]
row.names(ko.mgm.sub0) <- paste0(ko.mgm.sub0$KO,"_", ko.mgm.sub0$No, "_", ko.mgm.sub0$Level1, "_", ko.mgm.sub0$Level2 , "_", ko.mgm.sub0$Level3)

ko.sel <-ko.mgm.sub0[, c(-1:-6)]
ko.sel.ID <- ko.mgm.sub0[, c(1:6)]
colnames(ko.sel) == env.amp1$sample_name


env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.sel))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))
cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))
library(splitstackshape)

cor.out<-cSplit(cor.out, "Y", "_")

cor.out$Y_4 <- factor(cor.out$Y_4, levels = c("Bacterial.secretion.system","Cell.motility", "Xenobiotics.biodegradation.and.metabolism","Signal.transduction","Cellular.community.prokaryotes","Pentose.and.glucuronate.interconversions",
                                              "Translation",   "Energy.metabolism",  "Membrane.transport" , "Folding.sorting.and.degradation",
                                              "Metabolism.of.cofactors.and.vitamins"  ))

cor.out$Y_4 <- factor(cor.out$Y_4, labels  = c("BSS","CM", "XenoBDM","Signal transduction","Cellular community","PGI",
                                               "Translation",   "Energy metabolism",  "Membrane transport" , "FSD",
                                               
                                               "Metabolism of cofactors and vitamins"  ))

cor.out.sub<-cor.out[cor.out$Y_4=="Cellular community",]
levels(droplevels(factor(cor.out.sub$Y_5)))

p3<-ggplot(cor.out.sub, aes(Y_1,X)) +
  facet_grid(.~ Y_5,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 12,face="bold", color = c("white")),
        strip.background = element_rect(fill = "red"),
        panel.spacing = unit(0.1, "lines"), axis.title= element_blank(),
        axis.text.x=element_text(colour="black", size=4, face="bold",angle = 90, hjust=1),
        axis.text.y=element_text(colour="black", size=15, face="bold"))
p3
#ggsave("pdf2/Fig.S7c.corrHeatmap.env.Cellular.community.2023.01.05.pdf", width = 20, height = 6)
```


```{r fig.height=18, fig.width=20, message=FALSE, warning=FALSE}
#Supplementary Fig. 14
library(ggpubr)
ggarrange(p1,p2,p3,
          ncol = 1,nrow = 3,widths=c(1,1),heights=c(1,1),common.legend = TRUE,legend = "right",
          labels = c("a","b","c"))

#ggsave("pdf/Supplementary.Fig.13.pdf",plot = last_plot(),units = "in",height = 18,width = 20,dpi = 600)

```

```{r fig.height=7, fig.width=30, message=FALSE, warning=FALSE}
#Supplementary Fig. 15
ko.ID.sub<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "r.k_4")
ko.ID.sub <- merge(ko.ID.sub, ko.mgm.ID0, by.x = "KO", by.y = "Category", all = F)
ko.mgm1$KO <- row.names(ko.mgm1)
ko.mgm.sub0 <- merge(ko.ID.sub,ko.mgm1, by = "KO", all = T )
ko.mgm.sub0 <- ko.mgm.sub0[!is.na(ko.mgm.sub0$Level1),]
ko.mgm.sub0 <- ko.mgm.sub0[ order(ko.mgm.sub0$No),]
row.names(ko.mgm.sub0) <- paste0(ko.mgm.sub0$KO,"_", ko.mgm.sub0$No, "_", ko.mgm.sub0$Level1, "_", ko.mgm.sub0$Level2 , "_", ko.mgm.sub0$Level3)

ko.sel <-ko.mgm.sub0[, c(-1:-6)]
ko.sel.ID <- ko.mgm.sub0[, c(1:6)]
colnames(ko.sel) == env.amp1$sample_name


env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.sel))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))
cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

library(splitstackshape)

cor.out<-cSplit(cor.out, "Y", "_")

cor.out$Y_4 <- factor(cor.out$Y_4, levels = c("Bacterial.secretion.system","Cell.motility","Xenobiotics.biodegradation.and.metabolism","Signal.transduction","Metabolism.of.terpenoids.and.polyketides","Glycan.biosynthesis.and.metabolism","Porphyrin.metabolism",
                                              "Energy.metabolism","Membrane.transport","Citrate.cycle","Glyoxylate.and.dicarboxylate.metabolism","Metabolism.of.other.amino.acids"))

cor.out$Y_4 <- factor(cor.out$Y_4,labels = c("BSS","CM","XenoBDM","Two component system","MTP","GBM","PM",
                                             "Energy metabolism","Membrane transport","CC","GDM","MAC"))

cor.out.sub<-cor.out[cor.out$Y_4=="MTP",]
levels(droplevels(factor(cor.out.sub$Y_5)))

cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, levels = c( "Biosynthesis.of.12.14.and.16.membered.macrolides", "Biosynthesis.of.ansamycins.", "Biosynthesis.of.enediyne.antibiotics.", 
                                                       "Biosynthesis.of.siderophore.group.nonribosomal.peptides.", "Biosynthesis.of.type.II.polyketide.backbone.", "Biosynthesis.of.type.II.polyketide.products.", 
                                                       "Biosynthesis.of.vancomycin.group.antibiotics.", "Carotenoid.biosynthesis.", "Diterpenoid.biosynthesis.Including.Gibberellin.biosynthesis.", 
                                                       "Geraniol.degradation.", "Insect.hormone.biosynthesis.", "Limonene.and.pinene.degradation.",  
                                                       "Monoterpenoid.biosynthesis.", "Nonribosomal.peptide.structures.", 
                                                       "Polyketide.sugar.unit.biosynthesis.","Sesquiterpenoid.and.triterpenoid.biosynthesis.","Terpenoid.backbone.biosynthesis.", "Tetracycline.biosynthesis.", "Type.I.polyketide.structures.", 
                                                        "Zeatin.biosynthesis."))

cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, labels = c( "BMM", "Biosynthesis.of.ansamycins", "Biosynthesis.of.enediyne.antibiotics", 
                                                       "Biosynthesis.of.siderophore", "BPB", "Biosyn.polyketide", 
                                                       "Biosyn.vancomycin", "Carotenoid.biosynthesis", "D","Geraniol.D", 
                                                       "IH", "LPD", "M", "Nonribosomal.P","Polyketide.sugar.unit.biosynthesis", 
                                                       "STB", "Terpenoid.backbone.biosynthesis", 
                                                       "TB", "Type.I.polyketide.structures", "ZB"))

p1<-ggplot(cor.out.sub, aes(Y_1,X)) +
  facet_grid(.~ Y_5,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 10,face="bold", color = c("white")),
        strip.background = element_rect(fill = "red"),
        panel.spacing = unit(0.1, "lines"), axis.title= element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y=element_text(colour="black", size=18, face="bold")) +
  labs(caption="BBM: Biosynthesis.of.12.14.and.16.membered.macrolides;Biosynthesis.of.siderophore: Biosynthesis.of.siderophore.group.nonribosomal.peptides; BPB: Biosynthesis.of.type.II.polyketide.backbone; Biosyn.polyketide: Biosynthesis.of.type.II.polyketide.products; Biosyn.vancomycin: Biosynthesis.of.vancomycin.group.antibiotics; 
  D: Diterpenoid.biosynthesis.Including.Gibberellin.biosynthesis; Geraniol.D: Geraniol.degradation; IH: Insect.hormone.biosynthesis; LPD: Limonene.and.pinene.degradation; M: Monoterpenoid.biosynthesis; Nonribosomal.P: Nonribosomal.peptide.structures; TB: Terpenoid.backbone.biosynthesis; ZB: Zeatin.biosynthesis" )
p1
#ggsave("pdf2/Fig.S8a.corrHeatmap.env.Metabolism.Terpenoid.2023.01.05.pdf", width = 30, height = 7)

ko.ID.sub<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "r.k_4")
ko.ID.sub <- merge(ko.ID.sub, ko.mgm.ID0, by.x = "KO", by.y = "Category", all = F)
ko.mgm1$KO <- row.names(ko.mgm1)
ko.mgm.sub0 <- merge(ko.ID.sub,ko.mgm1, by = "KO", all = T )
ko.mgm.sub0 <- ko.mgm.sub0[!is.na(ko.mgm.sub0$Level1),]
ko.mgm.sub0 <- ko.mgm.sub0[ order(ko.mgm.sub0$No),]
row.names(ko.mgm.sub0) <- paste0(ko.mgm.sub0$KO,"_", ko.mgm.sub0$No, "_", ko.mgm.sub0$Level1, "_", ko.mgm.sub0$Level2 , "_", ko.mgm.sub0$Level3)

ko.sel <-ko.mgm.sub0[, c(-1:-6)]
ko.sel.ID <- ko.mgm.sub0[, c(1:6)]
colnames(ko.sel) == env.amp1$sample_name


env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.sel))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))
cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))
library(splitstackshape)

cor.out<-cSplit(cor.out, "Y", "_")

cor.out$Y_4 <- factor(cor.out$Y_4, levels = c("Bacterial.secretion.system","Cell.motility","Xenobiotics.biodegradation.and.metabolism","Signal.transduction","Metabolism.of.terpenoids.and.polyketides","Glycan.biosynthesis.and.metabolism","Porphyrin.metabolism",
                                              "Energy.metabolism","Membrane.transport","Citrate.cycle","Glyoxylate.and.dicarboxylate.metabolism","Metabolism.of.other.amino.acids"))

cor.out$Y_4 <- factor(cor.out$Y_4,labels = c("BSS","CM","XenoBDM","Two component system","MTP","GBM","PM",
                                             "Energy metabolism","Membrane transport","CC","GDM","MAC"))

cor.out.sub<-cor.out[cor.out$Y_4=="XenoBDM",]
levels(droplevels(factor(cor.out.sub$Y_5)))

cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, labels = c( "Aminobenzoate.degradation", "Atrazine.degradation", "Benzoate.degradation", 
                                                       "Bisphenol.degradation", "Caprolactam.degradation", "Chloroalkane.and.chloroalkene.degradation", 
                                                       "Chlorocyclohexane.and.chlorobenzene.degradation", "Dioxin.degradation", "Drug.metabolism.cytochrome.P450", 
                                                       "Drug.metabolism.other.enzymes", "Ethylbenzene.degradation", "Fluorobenzoate.degradation", 
                                                       "Furfural.degradation", "Metabolism.of.xenobiotics.by.cytochrome.P450",  
                                                       "Naphthalene.degradation", "Nitrotoluene.degradation", "Polycyclic.aromatic.hydrocarbon.degradation", 
                                                       "Steroid.degradation", "Styrene.degradation", "Toluene.degradation", 
                                                       "Xylene.degradation"))

cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, labels = c( "Aminobenzoate.degradation", "AtrD", "Benzoate.degradation", 
                                                       "B", "CapD", "ChloroalkaneD", 
                                                       "ChlcbD", "DioxinD", "Dc", 
                                                       "DrugO", "EbD", "FbD", 
                                                       "FfD", "Mxc", "NaD", 
                                                       "NiD", "PcahD", "SteroidD", 
                                                       "StyreneD", "TolueneD", "XyleneD"))

p2<-ggplot(cor.out.sub, aes(Y_1,X)) +
  facet_grid(.~ Y_5,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 10,face="bold", color = c("white")),
        strip.background = element_rect(fill = "red"),
        panel.spacing = unit(0.1, "lines"), axis.title= element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y=element_text(colour="black", size=18, face="bold")) +
  labs(caption="AtrD: Atrazine.degradation; B: Bisphenol.degradation; CapD: Caprolactam.degradation; ChlcbD: Chloroalkane.and.chloroalkene.degradation; ChlorocyclohexaneD: Chlorocyclohexane.and.chlorobenzene.degradation; DioxinD: Dioxin.degradation; Dc: Drug.metabolism.cytochrome.P450.; DrugO: Drug.metabolism.other.enzymes; EbD: Ethylbenzene.degradation; FbD: Fluorobenzoate.degradation; FfD: Furfural.degradation;
       Mxc: Metabolism.of.xenobiotics.by.cytochrome; NaD: Naphthalene.degradation; NiD: Nitrotoluene.degradation; PcahD: Polycyclic.aromatic.hydrocarbon.degradation; SD: Steroid.degradation; StyreneD: Styrene.degradation; TolueneD: Toluene.degradation; XyleneD: Xylene.degradation" )

p2
#ggsave("pdf2/Fig.S8b.corrHeatmap.env.XenoBDM.2023.01.05.pdf", width = 30, height = 7)

env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(rgi.mgm1))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))
cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))
library(splitstackshape)

cor.out.sig <- cor.out
cor.out.sig$r[cor.out.sig$p > 0.01] <- 0
cor.out.sig$Y_2<-"Antibiotic resistance genes (ARG)"

p3<-ggplot(cor.out.sig, aes(Y,X)) +
  facet_grid(.~ Y_2,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 10,face="bold", color = c("white")),
        strip.background = element_rect(fill = "red"),
        panel.spacing = unit(0.1, "lines"),
        axis.title= element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y=element_text(colour="black", size=18, face="bold"))
p3

#ggsave("pdf2/Fig.S8c.corrHeatmap.env.ARG.genes.2023.01.09.pdf", width = 30, height = 7)
```


```{r fig.height=21, fig.width=30, message=FALSE, warning=FALSE}
#Supplementary Fig. 15
library(ggpubr)
ggarrange(p1,p2,p3,
          ncol = 1,nrow = 3,widths=c(1,1),heights=c(1,1),common.legend = TRUE,legend = "right",
          labels = c("a","b","c"))

#ggsave("pdf/Supplementary.Fig.14.pdf",plot = last_plot(),units = "in",height = 21,width = 30,dpi = 600)

```

```{r fig.height=3, fig.width=10, message=FALSE, warning=FALSE}
#Supplementary Fig. 16
ko.ID.sub<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "r.k_4")
ko.ID.sub <- merge(ko.ID.sub, ko.mgm.ID0, by.x = "KO", by.y = "Category", all = F)
ko.mgm1$KO <- row.names(ko.mgm1)
ko.mgm.sub0 <- merge(ko.ID.sub,ko.mgm1, by = "KO", all = T )
ko.mgm.sub0 <- ko.mgm.sub0[!is.na(ko.mgm.sub0$Level1),]
ko.mgm.sub0 <- ko.mgm.sub0[ order(ko.mgm.sub0$No),]
row.names(ko.mgm.sub0) <- paste0(ko.mgm.sub0$KO,"_", ko.mgm.sub0$No, "_", ko.mgm.sub0$Level1, "_", ko.mgm.sub0$Level2 , "_", ko.mgm.sub0$Level3)

ko.sel <-ko.mgm.sub0[, c(-1:-6)]
ko.sel.ID <- ko.mgm.sub0[, c(1:6)]
colnames(ko.sel) == env.amp1$sample_name


env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.sel))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))
cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))
library(splitstackshape)

cor.out<-cSplit(cor.out, "Y", "_")

cor.out$Y_4 <- factor(cor.out$Y_4, levels = c("Bacterial.secretion.system","Cell.motility","Xenobiotics.biodegradation.and.metabolism","Signal.transduction","Metabolism.of.terpenoids.and.polyketides","Glycan.biosynthesis.and.metabolism","Porphyrin.metabolism",
                                              "Energy.metabolism","Membrane.transport","Citrate.cycle","Glyoxylate.and.dicarboxylate.metabolism","Metabolism.of.other.amino.acids"))

cor.out$Y_4 <- factor(cor.out$Y_4,labels = c("BSS","CM","XenoBDM","Two component system","MTP","GBM","PM",
                                             "Energy metabolism","Membrane transport","CC","GDM","MAC"))

cor.out.sub<-cor.out[cor.out$Y_4=="Energy metabolism",]
levels(droplevels(factor(cor.out.sub$Y_5)))
cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, labels = c("Carbon.fixation", "Methane.metabolism",  "Nitrogen.M",  "Oxidative.phosphorylation",   "Sulfur.metabolism" ))

p1<-ggplot(cor.out.sub, aes(Y_1,X)) +
  facet_grid(.~ Y_5,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 10,face="bold", color = c("white")),
        strip.background = element_rect(fill = "blue"),
        panel.spacing = unit(0.1, "lines"), axis.title= element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y=element_text(colour="black", size=8, face="bold")) +
  labs(caption="Nitrogen.M: Nitrogen metabolism")
p1
#ggsave("pdf2/Fig.S9a.corrHeatmap.env.Energy.metabolism.2023.01.05.pdf", width = 10, height = 3)

ko.ID.sub<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "r.k_4")
ko.ID.sub <- merge(ko.ID.sub, ko.mgm.ID0, by.x = "KO", by.y = "Category", all = F)
ko.mgm1$KO <- row.names(ko.mgm1)
ko.mgm.sub0 <- merge(ko.ID.sub,ko.mgm1, by = "KO", all = T )
ko.mgm.sub0 <- ko.mgm.sub0[!is.na(ko.mgm.sub0$Level1),]
ko.mgm.sub0 <- ko.mgm.sub0[ order(ko.mgm.sub0$No),]
row.names(ko.mgm.sub0) <- paste0(ko.mgm.sub0$KO,"_", ko.mgm.sub0$No, "_", ko.mgm.sub0$Level1, "_", ko.mgm.sub0$Level2 , "_", ko.mgm.sub0$Level3)

ko.sel <-ko.mgm.sub0[, c(-1:-6)]
ko.sel.ID <- ko.mgm.sub0[, c(1:6)]
colnames(ko.sel) == env.amp1$sample_name


env.tmp1<-env.amp1[,c("latitude","longitude","altitude","MAP","MAT",
                      "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                      "AllP_abu","AllP_bsa","AllP_rich")]

env.tmp2 <- data.frame(t(ko.sel))

spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)


r<-data.frame(spman.d12$r)
r[is.na(r)] <-0
r$X<-row.names(r)

r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
p<-data.frame(spman.d12$p)
p$X<-row.names(p)
p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
cor.out<-cbind(r.long,p.long$p)
cor.out$r <- round(as.numeric(cor.out$r), 2)
str(cor.out$r)

cor.out$X<- factor(cor.out$X, 
                   levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                              "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))

cor.out$X<- factor(cor.out$X, 
                   labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                              "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                              "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                              "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))

library(splitstackshape)

cor.out<-cSplit(cor.out, "Y", "_")

cor.out$Y_4 <- factor(cor.out$Y_4, levels = c("Bacterial.secretion.system","Cell.motility","Xenobiotics.biodegradation.and.metabolism","Signal.transduction","Metabolism.of.terpenoids.and.polyketides","Glycan.biosynthesis.and.metabolism","Porphyrin.metabolism",
                                              "Energy.metabolism","Membrane.transport","Citrate.cycle","Glyoxylate.and.dicarboxylate.metabolism","Metabolism.of.other.amino.acids"))

cor.out$Y_4 <- factor(cor.out$Y_4,labels = c("BSS","CM","XenoBDM","Two component system","MTP","GBM","PM",
                                             "Energy metabolism","Membrane transport","CC","GDM","MAC"))

cor.out.sub<-cor.out[cor.out$Y_4=="Membrane transport",]
levels(droplevels(factor(cor.out.sub$Y_5)))
cor.out.sub$Y_5 <- factor(cor.out.sub$Y_5, labels = c("ABC.transporters" , "PS" ))

p2<-ggplot(cor.out.sub, aes(Y_1,X)) +
  facet_grid(.~ Y_5,  scales = "free", space = "free" )+
  geom_tile(aes(fill = r), size=0)+
  scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
  geom_hline(yintercept = c(8.5,9.5), color = "orange")+
  theme(strip.text = element_text(size = 10,face="bold", color = c("white")),
        strip.background = element_rect(fill = "blue"),
        panel.spacing = unit(0.1, "lines"), axis.title= element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y=element_text(colour="black", size=8, face="bold")) +
  labs(caption="PS: Phosphotransferase.system")
p2
#ggsave("pdf2/Fig.S9b.corrHeatmap.env.Membrane.transport.2023.01.05.pdf", width = 10, height = 3)
```


```{r fig.height=6, fig.width=10, message=FALSE, warning=FALSE}
#Supplementary Fig. 16
library(ggpubr)
ggarrange(p1,p2,
          ncol = 1,nrow = 2,widths=c(1,1),heights=c(1,1),common.legend = TRUE,legend = "right",
          labels = c("a","b"))

#ggsave("pdf/Supplementary.Fig.15.pdf",plot = last_plot(),units = "in",height = 6,width = 10,dpi = 600)

```

```{r fig.height=6, fig.width=20, message=FALSE, warning=FALSE}
#Supplementary Fig. 17
rm(list = ls())
load("data/CForBio.data.prep.2022.11.22.RData")

library(vegan)
cutoff <- 1000

sort(colSums(bac.amp1))
sort(colSums(bac.mgm1))
sort(colSums(ko.mgm1))
sort(colSums(cog.mgm1))
sort(colSums(rgi.mgm1))
sort(colSums(cazy.mgm1))

bac.amp <- rrarefy(t(bac.amp1),sample = 24903 )
bac.mgm <- data.frame(t(bac.mgm1))

ko.mgm <- data.frame(t(ko.mgm1))
cog.mgm <- data.frame(t(cog.mgm1))
rgi.mgm <- data.frame(t(rgi.mgm1))
cazy.mgm <- data.frame(t(cazy.mgm1))

env.amp1$rgi.mgm.H <- vegan::diversity(rgi.mgm)
env.amp1$cazy.mgm.H <- vegan::diversity(cazy.mgm)
env.amp1$ko.mgm.H <- vegan::diversity(ko.mgm)
env.amp1$cog.mgm.H <- vegan::diversity(cog.mgm)
env.amp1$bac.mgm.H <- vegan::diversity(bac.mgm)
env.amp1$bac.amp.H <- vegan::diversity(bac.amp)

env.amp1$rgi.mgm.S <- vegan::specnumber(rgi.mgm)
env.amp1$cazy.mgm.S <- vegan::specnumber(cazy.mgm)
env.amp1$ko.mgm.S <- vegan::specnumber(ko.mgm)
env.amp1$cog.mgm.S <- vegan::specnumber(cog.mgm)
env.amp1$bac.mgm.S <- vegan::specnumber(bac.mgm)
env.amp1$bac.amp.S <- vegan::specnumber(bac.amp)

env.amp1$cog.mgm.sum <- rowSums(cog.mgm)
env.amp1$rgi.mgm.sum <- rowSums(rgi.mgm)
env.amp1$cazy.mgm.sum <- rowSums(cazy.mgm)
env.amp1$ko.mgm.sum <- rowSums(ko.mgm)
env.amp1$bac.mgm.sum <- rowSums(bac.mgm)

#func <- c("Ribosome","Aminoacyl.tRNA.biosynthesis")
#nam <- "Translation"
#hi <- 20
Heatmap.genes <- function (func, nam, hi){
  ko.ID<-read_excel("data/KEGG.123.2022.12.09.xlsx",sheet = "KEGG3")
  ko.ID.sub <- data.frame(ko.ID[ko.ID$Level3 %in%  func,])
  
  ko.mgm.sub <- ko.mgm1[row.names(ko.mgm1) %in% ko.ID.sub$KO,]
  ko.ID.sub <- ko.ID.sub[ko.ID.sub$KO %in% row.names(ko.mgm1) ,]
  ko.ID.sub <- ko.ID.sub[order(ko.ID.sub$KO),]
  row.names(ko.mgm.sub) == ko.ID.sub$KO
  ko.ID.sub$KO.id <- paste0(ko.ID.sub$Level3, "_",ko.ID.sub$KO)
  row.names(ko.mgm.sub) <- ko.ID.sub$KO.id
  
  ko.mgm.sub <- ko.mgm.sub[order(ko.ID.sub$No),]
  env.raw1<-env.amp1
  rownames(env.raw1)<-env.raw1$sample_name
  env.sel<-env.raw1[row.names(env.raw1) %in% colnames(ko.mgm.sub),]
  function.sel<-data.frame(t(ko.mgm.sub))
  row.names(env.sel) == row.names(function.sel)
  
  
  env.sel[, paste0(func, ".RA")] <- rowSums(function.sel)
  env.sel[, paste0(func, ".H")] <- vegan::diversity(function.sel)
  env.sel[, paste0(func, ".S")] <- vegan::specnumber(function.sel)
  cog.mgm <- data.frame(t(cog.mgm1))
  row.names(cog.mgm) == row.names(env.sel)
  env.sel$cog.mgm.sum <- rowSums(cog.mgm)
  env.cog <- data.frame(env.sel,cog.mgm )
  env.cog$rib.pct <- env.cog$J/env.cog$cog.mgm.sum
  env.cog$transcript.pct <- env.cog$K/env.cog$cog.mgm.sum
  env.cog$defense.pct <- env.cog$V/env.cog$cog.mgm.sum
  
  env.tmp1<-env.cog[,c("latitude","longitude","altitude","MAP","MAT",
                       "TC","TN","TP","pH","ACa","AMg","AFe","AK","C_N","C_P", "N_P", "soil_D",
                       "AllP_abu","AllP_bsa","AllP_rich")]
  
  env.tmp1<-data.frame(apply(env.tmp1,2,as.numeric))
  env.tmp2 <- function.sel
  
  spman.d12 = corr.test(env.tmp1, env.tmp2,use="pairwise",method="spearman",adjust="fdr",alpha=.05,ci=FALSE)
  
  
  r<-data.frame(spman.d12$r)
  r[is.na(r)] <-0
  r$X<-row.names(r)
  
  r.long <- gather(r, Y, r, 1 : ncol(r)-1,  factor_key=TRUE)
  p<-data.frame(spman.d12$p)
  p$X<-row.names(p)
  p.long <- gather(p, Y, p, 1 : ncol(p)-1, factor_key=TRUE)
  cor.out<-cbind(r.long,p.long$p)
  cor.out$r <- round(as.numeric(cor.out$r), 2)
  str(cor.out$r)
  
  cor.out$X<- factor(cor.out$X, 
                     levels = c("latitude","longitude","altitude","MAP","MAT","TC","TN","TP","pH","ACa","AMg","AFe",
                                "AK","C_N","C_P", "N_P", "soil_D","AllP_abu","AllP_bsa","AllP_rich"))
  
  cor.out$X<- factor(cor.out$X, 
                     labels = c("Latitude","Longitude","Altitude","Mean annual precipitation","Mean annual temperature",
                                "Total Carbon","Soil total Nitrogen","Soil total Phosphorus","Soil pH","Soil available Calcium",
                                "Soil available Magnesium", "Soil available Ferrum", "Soil available Potassium", "Carbon : Nitrogen ratio", "Carbon : Phosphorus ratio", "Nitrogen: Phosphorus ratio", 
                                "Soil bulk density", "Plant abundance", "Plant basal area", "Plant richness"))
  library(splitstackshape)
  #cor.out$Y 
  cor.out<-cSplit(cor.out, "Y", "_")
  cor.out$Y_1 <- factor(cor.out$Y_1, levels = func)
  
  p0<-ggplot(cor.out, aes(Y_2,X)) +
    facet_grid(cols = vars(Y_1), scales = "free", space = "free")+
    geom_tile(aes(fill = r), size=1)+
    geom_hline(yintercept = c(8.5,9.5), color = "red")+
    scale_fill_gradient(guide = "legend", high='green', low='blue',name="rho")+
    theme(axis.title= element_blank(),
          axis.text.x=element_text(colour="black", size=8, face="bold",angle = 90),
          axis.text.y=element_text(colour="black", size=12, face="bold"))
  #print(p0)
  
  cor.out.sig <-cor.out
  cor.out.sig$r[cor.out.sig$p > 0.05] <- 0
  
  p1<-ggplot(cor.out.sig, aes(Y_2,X)) + 
    facet_grid(cols = vars(Y_1), scales = "free", space = "free")+
    geom_tile(aes(fill = r), size=1)+
    geom_text(data=  cor.out.sig[cor.out.sig$r!=0,], aes(label = r), size = 1, font="bold")+scale_fill_gradient2(guide = "legend", high='#2c7bb6',mid="white", low='#d7191c',name="rho")+
   geom_hline(yintercept = c(8.5,9.5), color = "orange")+
    theme_bw()+
    theme(axis.title= element_blank(),
          strip.text = element_text(size = 12,face="bold", color = "white"),
          strip.background = element_rect(fill = "blue"),
          panel.spacing = unit(0, "lines"),
          axis.text.x=element_text(colour="black", size=8, face="bold",angle = 90,vjust = 0.5),
          axis.text.y=element_text(colour="black", size=12,face="bold"))
  
  #ggsave(paste0("pdf/Supplementary.Fig.11.Heatmap2.",nam,".2023.01.05.pdf"), width = hi, height = 6)
  print(p1)

}

Heatmap.genes(  c("Ribosome","Aminoacyl.tRNA.biosynthesis"), "Translation", 20  )

```


## R Markdown


